rpcperms/middleware_handler.go

```diff
@@ -529,3 +530,43 @@ func parseProto(typeName string, serialized []byte) (proto.Message, error) {
 
 	return msg.Interface(), nil
 }
+
+// replaceProtoMsg replaces the given target message with the content of the
+// replacement message.
+func replaceProtoMsg(target interface{}, replacement interface{}) error {
```

Comment:
Style nit: can now be `any` with Go 1.18.

Reply:
Or shoudl this just actually be `proto.Message` can we do the `protoiface` assertion later (if it's still needed?)

Reply:
Or could we maybe this this? https://pkg.go.dev/github.com/golang/protobuf/proto#Merge

Reply:
Or this https://pkg.go.dev/github.com/golang/protobuf/proto#UnmarshalMerge

Reply:
Okay, it turns out we don't need to call `Reset` specifically, that seems to be the default of `Unmarshal`. So I was able to simplify this a bit.